Update 04-upgrading-to-v1.4.md (#2710)
fishtown-analytics/docs.getdbt.com: website/docs/guides/migration/versions/04-upgrading-to-v1.4.md

### For maintainers of adapter plugins

> **TL;DR** Not much heavy lifting for this minor version. We anticipate more work for `1.5.0`. We plan to release betas early & often, and provide guidance on upgrading.

> The high-level changes are:
> - Add support for Python 3.11
> - Rename/Replace deprecated exception functions
> - Add support for Incremental Predicates (if applicable)
> - Make use of new adapter-zone tests
> 
> See [dbt-core/discussions/6624](https://github.com/dbt-labs/dbt-core/discussions/6624) for more detailed in for and to ask any questions you may have!

Reviewer: ```suggestion
The high-level changes are:
- Add support for Python 3.11
- Rename/replace deprecated exception functions
- Add support for Incremental Predicates (if applicable)
- Make use of new adapter-zone tests

For more detailed information and to ask any questions, please visit [dbt-core/discussions/6624](https://github.com/dbt-labs/dbt-core/discussions/6624).
```

### For maintainers of adapter plugins

> > **TL;DR** Not much heavy lifting for this minor version. But FYI there will be more work for `1.5.0`. Be warned!

Reviewer 2: Should we take a more proactive tone here? v1.5 will require more work, and here's how we plan to help. Just an idea:
```suggestion
> **TL;DR** Not much heavy lifting for this minor version. We anticipate more work for `1.5.0`. We plan to release betas early & often, and provide guidance on upgrading.
```

The high-level changes are:
- Add support for Python 3.11
- Rename/Replace deprecated exception functions

> - Consolidate timestamp functions & macros

Reviewer 2: I think this change was actually included in v1.3. It landed in the v1.4 changelog because it was merged during v1.3 RC period

> - Consolidate timestamp functions & macros

Reviewer: ```suggestion
```
